bee-hive/tests/workflow/test_sequence.py

```diff
@@ -0,0 +1,40 @@
+import unittest
+from unittest.mock import MagicMock
+from bee_hive.workflow import Workflow
+
+class TestSequenceMethod(unittest.TestCase):
+    def setUp(self):
+        self.mock_workflow = {
+            "spec": {
+                "prompt": "Start of the workflow",
+                "steps": [
+                    {"name": "step1", "agent": "agent1"},
+                    {"name": "step2", "agent": "agent2"},
+                    {"name": "step3", "agent": "agent1"}
+                ]
```

Comment:
I think steps here would also need to move under `spec.template` instead of just `spec`